### DISCUSSION TITLE
Strapi `3.1.3` breaks `upload-file` relationships

### DISCUSSION THREAD

**Discussion by cupofnestor** [strapi/strapi#7475] [Legacy Topics]
We have ±20 models, many with files attached.  I have migrated these across  many pre-release versions of strapi in my own docker setup. 

This is my method:

1. Generate new project using latest version
2. mongorestore previous database
3. move components and models
4. move uploads
5. build admin using strapi develop

Now, in the latest upgrade from `v3.0.0-beta.19.3` to `v3.1.3` we have lost all of the relationships between `upload-files` and our models.  The files themselves are there, they appear in the database, and the media browser. They are not appearing in our models, or queries.

The _initial_ database clone has the `related` relationships intact.
<img width="406" alt="Screen Shot 2020-08-14 at 9 54 44 AM" src="https://user-images.githubusercontent.com/550422/90256801-3bab0f80-de14-11ea-910c-d42917b1946d.png">
This is identical to the original database.

Once `strapi develop` is run (connected to this database), the `related` array seems to be stripped.

<img width="416" alt="Screen Shot 2020-08-14 at 9 56 14 AM" src="https://user-images.githubusercontent.com/550422/90256921-6d23db00-de14-11ea-8d4d-b96d7e672820.png">

While I understand my use and upgrade method is a bit unconventional, this seems like it may be a bug as it produces some unexpected behaviour inconsistent with previous versions.  I'd love to use the release version in production, but due to the volume of media content, manually re-linking may be a non-starter.

I have not tested whether re-populating the array "breaks" anything, but will do that next.

**Comment by lauriejim** [comment]
Hello! Direct migration from the beta.19 to 3.1.3 is not recommended.
You have to follow each migration guides one by one from the beta.19 to 3.1.3.
So first - beta.19 to beta.20 - https://strapi.io/documentation/v3.x/migration-guide/migration-guide-beta.19-to-beta.20.html
Then - beta.20 to stable 3.0.0 - https://strapi.io/documentation/v3.x/migration-guide/migration-guide-beta.20-to-3.0.0.html
And finally - 3.0.0 to 3.1.0 - https://strapi.io/documentation/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.html

**Reply by derrickmehaffy** [reply]
Specifically for mongo users you will need this guide: https://strapi.io/documentation/v3.x/migration-guide/migration-guide-beta.19-to-beta.20.html#export-model-metadatas

(For anyone else that finds this in a search :wink: )

**Reply by cupofnestor** [reply]
🙏🏼
One thing to note is that the `migration.js` doesn't know about your database when you run in mongo shell.

```
>use strapi
>load('migration.js')
```

The load() method takes a path argument, and some/most versions of mongo default to the `dbpath` ( /data/db ) as the default location.
